compliance-monitor/monitor.py

```diff
@@ -421,36 +430,42 @@ async def post_report(
     sep += len(SEP)
     signature = body_text[:sep - 1]  # do away with the ampersand!
     body_text = body_text[sep:]
-    json_text = None
+    try:
+        ssh_validate(keys, signature, body_text)
+    except ValueError:
+        raise HTTPException(status_code=401)
+
+    json_texts = []
     if content_type.endswith('-yaml'):
         yaml = ruamel.yaml.YAML(typ='safe')
-        document = yaml.load(body_text)
-        json_text = json.dumps(document, cls=TimestampEncoder)
+        documents = yaml.load_all(body_text)
+        json_texts = [json.dumps(document, cls=TimestampEncoder) for document in documents]
     elif content_type.endswith("-json"):
-        document = json.loads(body_text)
-        json_text = body_text
+        documents = [json.loads(body_text)]
+        json_texts = [body_text]
     else:
         raise HTTPException(status_code=500)
-    rundata = document['run']
-    uuid, subject, checked_at = rundata['uuid'], document['subject'], document['checked_at']
-    check_role(account, subject, ROLES['append_any'])
-    with conn.cursor() as cur:
-        keys = db_get_keys(cur, subject)
-    try:
-        ssh_validate(keys, signature, body_text)
-    except ValueError:
-        raise HTTPException(status_code=401)
-    scopeuuid = document['spec']['uuid']
+
+    allowed_subjects.add(account[0])
```

Comment:
This should be moved to where `allowed_subjects` is initially created, or `db_find_subjects(cursor, X)` always should return a list containing at least `X`.

Reply:
I will do something about that. I think the call to db_find_subjects should just store the result in a list, and it should be clear that this list is NOT the list of allowed subjects, but the list of delegations or something. And only here would I construct the set using both this list and the login info.

Reply:
9f10620785229d67c4f6cc39b303fd829445d4d3